x/gamm/types/route.go

```diff
@@ -1,9 +1,16 @@
 package types
 
-import sdk "github.com/cosmos/cosmos-sdk/types"
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	appparams "github.com/osmosis-labs/osmosis/v11/app/params"
+)
 
 type SwapAmountInRoutes []SwapAmountInRoute
 
+func (routes SwapAmountInRoutes) IsOsmoRoutedMultihop() bool {
+	return len(routes) == 2 && (routes[0].TokenOutDenom == appparams.BaseCoinUnit)
```

Comment:
Two questions here:
1. Why do we have len == 2 here? What if route has length of three which all includes Osmo in the routes?
2. Why are we always taking 0 index of routes and expect it to be basecoin unit here?

Reply:
In general the idea that osmo will have a lot of pools X-OSMO
Then to Swap X->Y  all swaps will looks like X -> OSMO -> Y
So to detect such swaps we take the first hop and check that it has output OSMO(that mean that input for the second hop will be osmo too)

What if route has length of three which all includes Osmo in the routes 
- That possible only if we have second hot OSMO->OSMO, or i miss something?